bundles/redhat8.6/bundle.sh.gotmpl

```diff
@@ -13,57 +13,94 @@ RHSM_ACTIVATION_KEY=${RHSM_ACTIVATION_KEY:-""}
+  if [[ -n "${SATELLITE_SERVER_URL}" ]]; then
+    rpm -ivh "${SATELLITE_SERVER_URL}/pub/katello-ca-consumer-latest.noarch.rpm"
+  fi
 
-if [[ ${FOUND} == false ]]; then
-  echo "You must use subscription manager to fetch packages for redhat"
-  exit 1
-fi
+  if [[ -n "${RHSM_ORG_ID}" && -n "${RHSM_ACTIVATION_KEY}" ]]; then
+    subscription-manager register --org="${RHSM_ORG_ID}" --activationkey="${RHSM_ACTIVATION_KEY}" --force
+    FOUND=true
+  fi
+
+  if [[ -n "${RHSM_USER}" && -n "${RHSM_PASS}" && ${FOUND} == false ]]; then
+    subscription-manager register --username="${RHSM_USER}" --password="${RHSM_PASS}" --force
+    FOUND=true
+  fi
+
+  if [[ ${FOUND} == false ]]; then
+    echo "You must use subscription manager to fetch packages for redhat"
+    exit 1
+  fi
 
-subscription::unregister() {
-  subscription-manager unregister
-}
+  subscription::unregister() {
+    subscription-manager unregister
+  }
 
-subscription::defer_unregister() {
+  subscription::defer_unregister() {
     trap subscription::unregister ABRT
     trap subscription::unregister EXIT
     trap subscription::unregister HUP
     trap subscription::unregister INT
     trap subscription::unregister TERM
     trap subscription::unregister USR1
     trap subscription::unregister USR2
-}
+  }
+
+  subscription-manager release --set=8.6
+  subscription-manager refresh
+  subscription::defer_unregister
+
+  ENABLED_REPOS="kubernetes,codeready-builder-for-rhel-8-x86_64-rpms,rhel-8-for-x86_64-appstream-rpms,rhel-8-for-x86_64-baseos-rpms"
+  EUS_REPOS=${EUS_REPOS:-""}
+  if [[ -n "${EUS_REPOS}" ]]; then
+    #disables the standard repositories which should not be enabled when using EUS
+    subscription-manager repos --disable=rhel-8-for-x86_64-baseos-rpms --disable=rhel-8-for-x86_64-appstream-rpms
+    subscription-manager repos --enable rhel-8-for-x86_64-baseos-eus-rpms
+    subscription-manager repos --enable rhel-8-for-x86_64-appstream-eus-rpms
+    subscription-manager repos --enable codeready-builder-for-rhel-8-x86_64-eus-rpms
+    ENABLED_REPOS="kubernetes,codeready-builder-for-rhel-8-x86_64-eus-rpms,rhel-8-for-x86_64-appstream-eus-rpms,rhel-8-for-x86_64-baseos-eus-rpms"
+  else
+    subscription-manager repos --enable codeready-builder-for-rhel-8-x86_64-rpms
+    subscription-manager repos --enable rhel-8-for-x86_64-appstream-rpms
+    subscription-manager repos --enable rhel-8-for-x86_64-baseos-rpms
+  fi
+else
+  echo "Bypassing subscription-manager and upstream RHEL repositories due to --skip-subscription-manager being set to true."
+fi
+
+if [[ -f /etc/yum.repos.d/user-repos.repo ]]; then
+  USER_REPOS="$(awk -F '[][]' '/^\[.*\]/ {print $2}' /etc/yum.repos.d/user-repos.repo | paste -sd, -)"
+  if [[ -n "${ENABLED_REPOS}" ]] && [[ -n "${USER_REPOS}" ]]; then
+      ENABLED_REPOS="${ENABLED_REPOS},${USER_REPOS}"
+  elif [[ -z "${ENABLED_REPOS}" ]] && [[ -n "${USER_REPOS}" ]]; then
+      ENABLED_REPOS="${USER_REPOS}"
+  fi
+fi
```

Comment:
This requires users to define their rpm repos in `repo-templates/user-repos.repo`. We will need to document this workflow.

While KIB CLI automatically adds any repo files in `repo-templates/` to `/etc/yum.repos.d`, we need to selectively enable/disable repos that already exist there (e.g., `/etc/yum.repos.d/ubi.repo`). We could only parse files other than this one, but maybe it is simpler to just have one file that users modify and we look for.

Open to any feedback.